### DISCUSSION TITLE
Zephyr Does not Appear To Respect my_board.yaml

### DISCUSSION THREAD

**Discussion by markovejnovic** [zephyrproject-rtos/zephyr#73133] [Q&A] [answered]
Hey everyone! I am currently trying to create a new board in zephyr. I have designed a custom PCBA and I am now attempting to write the firmware for it. I am quite new to zephyr (coming from FRTOS) so this is a new-ish experience for me.

I am trying to build my `app/` out-of-tree and have added it to my workspace. Currently, my app is empty as I am trying to setup the basics for my board. I can confirm that 
```bash
west build -p always -b nucleo_f412zg
```
successfully builds my program.

I have the following file structure (my board's name is `biologger`)
```
.
└── app/
    ├── boards/arm/biologger/
    │   ├── Kconfig.biologger
    │   ├── Kconfig.defconfig
    │   ├── biologger.dts
    │   ├── biologger.yaml
    │   ├── biologger_defconfig
    │   └── board.yml
    ├── src/
    │   └── main.c - empty
    ├── CMakeLists.txt - only calls find_package(Zephyr)
    ├── VERSION - minimal
    └── prj.conf - empty
```

When I try to run west to build this board:
```bash
west build -p always -b biologger -- -DBOARD_ROOT=.
```
it fails on me saying that `FLASH` and `RAM` both overflowed. I looked into `autoconf.h` and noticed
```c
#define CONFIG_SRAM_SIZE 0
#define CONFIG_SRAM_BASE_ADDRESS 0x0
```
which would explain why my app overflows! :laughing: 

But... the problem now is that I can't seem to figure out why `autoconf.h` doesn't have the specified memory. My `biologger.yaml` is:
```yaml
identifier: biologger
name: BioLogger
type: mcu
arch: arm
toolchain:
  - zephyr
  - gnuarmemb
  - xtools
ram: 256
flash: 1024
supported:
  - i2c
  - gpio
  - usb_device
vendor: other
```

so I would imagine that the build system should respect my `256K` of ram and `1024K` of flash.

Comparing this to `boards/st/nucleo_f412zg/`, it appears there is no explicit case where `nucleo` also sets the kconfig `CONFIG_SRAM_SIZE` field... Yet somehow its `autoconf.h` does have the correct value:
```c
#define CONFIG_SRAM_SIZE 256
#define CONFIG_SRAM_BASE_ADDRESS 0x20000000
```

I therefore inferred that the `.yaml` is where the build system should attempt to pull this information from.

I have a representative repo [here](https://github.com/markovejnovic/biologger-fw).
The zephyr repo hash is c211cb347e0af0a4931e0e7af3d93577bcc7af8f

Thank you all for zephyr! Apologies if my question is wasteful of your time -- I couldn't find anything in issues/discussions.

Thanks!

**Comment by adri1mart1** [comment, marked as answer]
Hello,

I think you are just missing 

```
/ {
    chosen {
        zephyr,sram = &sram0;
    };
};
```

See [link](https://github.com/zephyrproject-rtos/zephyr/blob/c211cb347e0af0a4931e0e7af3d93577bcc7af8f/arch/Kconfig#L202)

I advise you to use all zephyr chosen "aliases" so that you can easily compile and test zephyr examples. [link](https://github.com/zephyrproject-rtos/zephyr/blob/c211cb347e0af0a4931e0e7af3d93577bcc7af8f/boards/st/nucleo_f412zg/nucleo_f412zg.dts#L17)

Regards,
Adrien M.

**Reply by markovejnovic** [reply]
Hi Adrien! Thank you so much! That did the trick! Appreciate your help very much!

Thanks a bunch, very much appreciated!

- Marko
